ktor-client/ktor-client-cio/jvmAndPosix/src/io/ktor/client/engine/cio/Endpoint.kt

```diff
@@ -282,7 +288,7 @@ internal class Endpoint(
     }
 
     private fun releaseConnection() {
-        val address = InetSocketAddress(host, port)
+        val address = lastResolvedAddress ?: return
```

Comment:
Not sure if `lastResolvedAddress` can be `null`. If not, we probably should throw an exception here